functors_task_2/student/include/impl.h

```diff
@@ -65,7 +65,14 @@ class DataBrowser
 
         const auto& it = m_dataReaders.find(userId);
 
-        return m_dataReaders.cend() == it || !it->second ? false : f(it->second);
+        bool result = false;
```

Comment:
please avoid one return rule
https://isocpp.github.io/CppCoreGuidelines/CppCoreGuidelines#nr2-dont-insist-to-have-only-a-single-return-statement-in-a-function